gsrs-rest-api/src/main/java/gsrs/api/GsrsEntityRestTemplate.java

```diff
@@ -92,24 +99,29 @@ public ExistsCheckResult exists(String... anyKindOfIdString) throws IOException{
                list.add(s);
            }
        }
-
-        return restTemplate.postForObject(prefix+"/@exists", list, ExistsCheckResult.class);
+       return restTemplate.postForObject(prefix+"/@exists", list, ExistsCheckResult.class);
     }
 
-    public <S extends T> Optional<PagedResult<S>> page(long top, long skip) throws JsonProcessingException {
+    public <S extends T> Optional<PagedResult<S>> page(long top, long skip) throws IOException {
         ResponseEntity<String> response = doGet("/?top=" + top +"&skip=" + skip,String.class);
-        if(response.getStatusCodeValue() == 404) {
+        if(response.getStatusCode() == HttpStatus.NOT_FOUND){
             return Optional.empty();
+        } else if(response.getStatusCode().is2xxSuccessful()){
+            try {
+                JsonNode node = mapper.readTree(response.getBody());
+                WithoutContentPagedResult result = mapper.convertValue(node, WithoutContentPagedResult.class);
+                JsonNode array = node.get("content");
+                if (array.isArray()) {
+                    List<S> content = parseFromJsonList(array);
+                    return Optional.of(result.toPagedResult(content));
+                }
+                return Optional.of(result.toPagedResult(Collections.emptyList()));
+            }catch(Exception ex) {
```

Comment:
should we catch Exception or Throwable ?  any reason NOT to catch Throwable ?

Reply:
According to this article throwable would catch errors in addition to exceptions.  It recommends catching specific exceptions/errors as possible. But that's not as easy as it sounds.  

https://www.baeldung.com/java-catch-throwable-bad-practice

If I am reading your RestTemplateResponseErrorHandler code right, I think you have any exception caused by SERVER_ERRROR or CLIENT_ERROR in the response turned OFF.  Anything other than those causes would still cause an exception/error to be raised.  In cases where we are now try-catching exceptions (Exception ex), **errors** would not be communicated to the user/client. The code would continue to execute.     

Personally, I usually care most that an error/exception has happened, and rely on the associated message to understand what it was.  Maybe if there are specific expected errors/exceptions one can try to catch those first and take selective action. Then one could catch a general throwable in case anything was missed.

But if one did not want to do all that selective work, one could just catch throwable in case anything at all went wrong. That way no error/exception would go unnoticed in the client.  

In brief I don't have a good reason not to use throwable.